### PR TITLE
Menus Simplified

### PR DESCRIPTION
**Pull request by Aericio**
Please make sure your issue complies with these guidelines:
- * [x] Use same formatting
- * [x] Changes must have been tested on PMMP.
- * [x] Unless it is a minor code modification, you must use an IDE.
- * [x] Have a detailed title (i.e "Fix Auctioned items remain in inventory")

### What does the PR change?
bruh

### Testing Environment
no

### CODE REVIEW

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctionBrowserMenu.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+use pocketmine\item\Item;
+use pocketmine\nbt\tag\IntTag;
+use pocketmine\Player;
+use pocketmine\scheduler\ClosureTask;
+use pocketmine\scheduler\TaskHandler;
+
+class AuctionBrowserMenu extends Menu
+{
+    const PAGE_LENGTH = self::PAGE_ROW_LENGTH * 4;
+    const PAGE_ROW_LENGTH = 7;
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/AuctionBrowserMenu.php]: Why not make these configurable?

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctionBrowserMenu.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        }
+    }
+
+    public function handle(Player $player, Item $itemClicked, Item $itemClickedWith, SlotChangeAction $action): bool
+    {
+        if ($itemClicked->getNamedTagEntry("AuctionID") !== null) {
+            $auction = PiggyAuctions::getInstance()->getAuctionManager()->getAuction(($itemClicked->getNamedTagEntry("AuctionID") ?? new IntTag())->getValue());
+            if ($auction instanceof Auction) {
+                $browserAttributes = [$player, $this->page, $this->search, $this->sortType];
+                new AuctionMenu($player, $auction, function () use ($browserAttributes) {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/AuctionBrowserMenu.php]: Use of a ``static function`` would be better here

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctionManagerMenu.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        $this->menu->getInventory()->setItem(22, Item::get(Item::ARROW)->setCustomName(PiggyAuctions::getInstance()->getMessage("menus.back")));
+        $this->menu->getInventory()->setItem(24, Item::get(Item::GOLDEN_HORSE_ARMOR)->setCustomName(PiggyAuctions::getInstance()->getMessage("menus.auction-manager.create-auction")));
+
+        $sort = $this->menu->getInventory()->getItem(23)->setCustomName(PiggyAuctions::getInstance()->getMessage("menus.sorting.sort-type", ["{TYPES}" => implode("\n", array_map(function (string $type, int $index): string {
+            return ($index === $this->sortType ? PiggyAuctions::getInstance()->getMessage("menus.sorting.selected") : "") . PiggyAuctions::getInstance()->getMessage("menus.sorting." . $type);
+        }, self::SORT_TYPES, array_keys(self::SORT_TYPES)))]));
+        $this->menu->getInventory()->setItem(23, $sort);
+
+        $auctions = array_filter(PiggyAuctions::getInstance()->getAuctionManager()->getAuctionsHeldBy($this->player), function (Auction $auction): bool {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/AuctionManagerMenu.php]: A ``static function`` can be used here

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctionManagerMenu.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                $this->sortType = array_keys(self::SORT_TYPES)[($key + 1) % 3];
+                $this->render();
+                break;
+            case 24:
+                new AuctionCreatorMenu($player);
+                break;
+            default:
+                $managerAttributes = [$player, $this->sortType];
+                $auction = PiggyAuctions::getInstance()->getAuctionManager()->getAuction(($itemClicked->getNamedTagEntry("AuctionID") ?? new IntTag())->getValue());
+                if ($auction !== null) new AuctionMenu($player, $auction, function () use ($managerAttributes) {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/AuctionManagerMenu.php]: A ``static function`` can be used here

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctionMenu.php [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        }), 20);
+        parent::__construct($player);
+    }
+
+    public function render(): void
+    {
+        $this->menu->setName(PiggyAuctions::getInstance()->getMessage("menus.auction-view.title"));
+
+        $this->menu->getInventory()->setItem(13, MenuUtils::getDisplayItem($this->auction));
+        $this->menu->getInventory()->setItem(33, Item::get(Item::MAP)->setCustomName(count($this->auction->getBids()) === 0 ? PiggyAuctions::getInstance()->getMessage("menus.auction-view.no-bids") : PiggyAuctions::getInstance()->getMessage("menus.auction-view.bid-history", ["{BIDS}" => count($this->auction->getBids()), "{HISTORY}" => implode("\n", array_map(function (AuctionBid $auctionBid): string {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/AuctionMenu.php]: A ``static function`` can be used here

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctioneerMenu.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            $this->render();
+        }), 20);
+        parent::__construct($player);
+    }
+
+    public function handle(Player $player, Item $itemClicked, Item $itemClickedWith, SlotChangeAction $action): bool
+    {
+        $auctioneer = $this->auctioneer;
+        $auction = PiggyAuctions::getInstance()->getAuctionManager()->getAuction(($itemClicked->getNamedTagEntry("AuctionID") ?? new IntTag())->getValue());
+        if ($auction !== null) new AuctionMenu($player, $auction, function () use ($player, $auctioneer) {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/AuctioneerMenu.php]: A ``static function`` can be used here

File: src/DaPigGuy/PiggyAuctions/menu/pages/BidsMenu.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        $this->taskHandler = PiggyAuctions::getInstance()->getScheduler()->scheduleRepeatingTask(new ClosureTask(function (int $currentTick): void {
+            $this->render();
+        }), 20);
+        parent::__construct($player);
+    }
+
+    public function render(): void
+    {
+        $this->menu->setName(PiggyAuctions::getInstance()->getMessage("menus.view-bids.title"));
+        $auctions = array_filter(array_map(function (AuctionBid $bid): ?Auction {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/BidsMenu.php]: A ``static function`` can be used here

File: src/DaPigGuy/PiggyAuctions/menu/pages/BidsMenu.php [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+    public function handle(Player $player, Item $itemClicked, Item $itemClickedWith, SlotChangeAction $action): bool
+    {
+        switch ($action->getSlot()) {
+            case 22:
+                new MainMenu($player);
+                break;
+            default:
+                $auction = PiggyAuctions::getInstance()->getAuctionManager()->getAuction(($itemClicked->getNamedTagEntry("AuctionID") ?? new IntTag())->getValue());
+                if ($auction !== null) new AuctionMenu($player, $auction, function () use ($player) {
```
Comment (TwistedAsylumMC) [inline src/DaPigGuy/PiggyAuctions/menu/pages/BidsMenu.php]: A ``static function`` can be used here

File: src/DaPigGuy/PiggyAuctions/menu/pages/AuctionBrowserMenu.php (hunk 1 above)
Comment (DaPigGuy) [reply src/DaPigGuy/PiggyAuctions/menu/pages/AuctionBrowserMenu.php]: These are for the browser borders

**Pull request merged**